monero/docker-compose.yml

```diff
@@ -0,0 +1,86 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: monero_server_1
+      APP_PORT: 9976
+      PROXY_AUTH_ADD: "true"
+
+  server:
+    image: deverickapollo/umbrel-monero:v1.0.0@sha256:a5ca2854b71de83a26a4176bf82f92609b0e67f54be5b7eb9386acffba87b33e
+    depends_on: [monerod]
+    restart: on-failure
+    expose:
+      - "9976"
+    volumes:
+      - ${APP_DATA_DIR}/data/app:/data # volume to persist advanced settings json
+      - ${APP_MONERO_DATA_DIR}:/monero/.monero # volume to persist bitmonero.conf
+    environment:
+      PORT: "9976"
+      MONERO_HOST: "${APP_MONERO_NODE_IP}"
+      MONERO_P2P_PORT: "${APP_MONERO_P2P_PORT}"
+      MONERO_RPC_PORT: "${APP_MONERO_RPC_PORT}"
+      MONERO_DEFAULT_NETWORK: "${MONERO_DEFAULT_NETWORK}"
+      MONERO_RPC_USER: "${APP_MONERO_RPC_USER}"
+      MONERO_RPC_PASSWORD: "${APP_MONERO_RPC_PASS}"
+      RPC_PORT: "${APP_MONERO_RPC_PORT}"
+      RPC_USER: "${APP_MONERO_RPC_USER}"
+      RPC_PASSWORD: "${APP_MONERO_RPC_PASS}"
+      MONERO_RPC_HIDDEN_SERVICE: "${APP_MONERO_RPC_HIDDEN_SERVICE}"
+      MONERO_P2P_HIDDEN_SERVICE: "${APP_MONERO_P2P_HIDDEN_SERVICE}"
+      DEVICE_DOMAIN_NAME: "${DEVICE_DOMAIN_NAME}"
+      MONEROD_IP: "${APP_MONERO_NODE_IP}"
+      TOR_PROXY_IP: "${APP_MONERO_TOR_PROXY_IP}"
+      TOR_PROXY_PORT: "9050"
+      TOR_PROXY_CONTROL_PORT: "9051"
+      TOR_PROXY_CONTROL_PASSWORD: "moneyprintergobrrr"
+      I2P_DAEMON_IP: "${APP_MONERO_I2P_DAEMON_IP}"
+      I2P_DAEMON_PORT: "7656"
+    networks:
+      default:
+        ipv4_address: $APP_MONERO_IP
+
+  monerod:
+    user: "1000:1000"
+    restart: unless-stopped
+    stop_grace_period: 15m
+    command: "${APP_MONERO_COMMAND}"
+    healthcheck:
+      disable: true
+    image: sethsimmons/simple-monerod:v0.18.2.2@sha256:835a9305b02c0a640363ddc2e77d709289ff54c402ede0a7ca242ab70b6270d4
+    ports:
+      - "${APP_MONERO_P2P_PORT}:${APP_MONERO_P2P_PORT}"
+      - "${APP_MONERO_RPC_PORT}:${APP_MONERO_RPC_PORT}"
+    expose:
+      - "${APP_MONERO_P2P_PORT}"
+      - "${APP_MONERO_RPC_PORT}"
```

Comment:
Did you find that exposing the ports was necessary? I think you should be okay just binding the ports as you have done above.